drivers/serial/uart_intel_lw.c

```diff
@@ -0,0 +1,999 @@
+	} while ((status & INTEL_LW_UART_STATUS_TRDY_MSK) == 0);
+
+	sys_write32(c, config->base + INTEL_LW_UART_TXDATA_REG_OFFSET);
+
+	k_spin_unlock(&data->lock, key);
+}
+
+/**
+ * @brief Initialise an instance of the driver
+ *
+ * This function initialise the interrupt configuration for the driver.
+ *
+ * @param dev UART device instance
+ *
+ * @return 0 to indicate success.
+ */
+static int uart_intel_lw_init(const struct device *dev)
+{
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	struct uart_intel_lw_device_data *data = dev->data;
+	const struct uart_intel_lw_device_config *config = dev->config;
+	/* clear status to ensure, that interrupts are not triggered due to old status. */
+	sys_write32(INTEL_LW_UART_CLEAR_STATUS_VAL, config->base
+				+ INTEL_LW_UART_STATUS_REG_OFFSET);
+
+	/*
+	 * Enable hardware interrupt.
+	 * The corresponding csr from IP still needs to be set,
+	 * so that the IP generates interrupt signal.
+	 */
+	config->irq_config_func(dev);
+
+#ifdef CONFIG_UART_LINE_CTRL
+	/* Enable DCTS interrupt. */
+	data->control_val = INTEL_LW_UART_CONTROL_DCTS_MSK;
+#endif /* CONFIG_UART_LINE_CTRL */
+
+	sys_write32(data->control_val, config->base + INTEL_LW_UART_CONTROL_REG_OFFSET);
+
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+	return 0;
+}
+
+/**
+ * @brief Check if an error was received
+ * If error is received, it will be mapped to uart_rx_stop_reason.
+ * This function should be called after irq_update.
+ * If interrupt driven API is not enabled,
+ * this function will read and clear the status register.
+ *
+ * @param dev UART device struct
+ *
+ * @return UART_ERROR_OVERRUN, UART_ERROR_PARITY, UART_ERROR_FRAMING,
+ * UART_BREAK if an error was detected, 0 otherwise.
+ */
+static int uart_intel_lw_err_check(const struct device *dev)
+{
+	struct uart_intel_lw_device_data *data = dev->data;
+	int err = UART_INTEL_LW_NO_ERROR;
+
+#ifndef CONFIG_UART_INTERRUPT_DRIVEN
+	const struct uart_intel_lw_device_config *config = dev->config;
+	k_spinlock_key_t key = k_spin_lock(&data->lock);
+
+	data->status_act = sys_read32(config->base + INTEL_LW_UART_STATUS_REG_OFFSET);
+#endif
+
+	if (data->status_act & INTEL_LW_UART_STATUS_E_MSK) {
+		if (data->status_act & INTEL_LW_UART_STATUS_PE_MSK) {
+			err |= UART_ERROR_PARITY;
+		}
+
+		if (data->status_act & INTEL_LW_UART_STATUS_FE_MSK) {
+			err |= UART_ERROR_FRAMING;
+		}
+
+		if (data->status_act & INTEL_LW_UART_STATUS_BRK_MSK) {
+			err |= UART_BREAK;
+		}
+
+		if (data->status_act & INTEL_LW_UART_STATUS_ROE_MSK) {
+			err |= UART_ERROR_OVERRUN;
+		}
+	}
+
+#ifndef CONFIG_UART_INTERRUPT_DRIVEN
+	/* clear status */
+	sys_write32(INTEL_LW_UART_CLEAR_STATUS_VAL, config->base
+	+ INTEL_LW_UART_STATUS_REG_OFFSET);
+	k_spin_unlock(&data->lock, key);
+#endif
+
+	return err;
+}
+
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+/***
+ * @brief helper function to check, if the configuration is support.
+ * @param  cfg_stored : The original configuration.
```

Comment:
Remove colon (`:`) or else it would be in the parameter description.

Reply:
done